runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/validator/StringValidator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.binding.kafka.internal.validator;
+
+import io.aklivity.zilla.runtime.binding.kafka.internal.types.OctetsFW;
+import io.aklivity.zilla.runtime.binding.kafka.internal.validator.config.StringValidatorConfig;
+
+public class StringValidator implements Validator
+{
+    private String encoding;
+
+    public StringValidator(
+        StringValidatorConfig config)
+    {
+        this.encoding = config.encoding;
+    }
+
+    @Override
+    public boolean validate(
+        OctetsFW data)
+    {
+        boolean valid = true;
+        if (data != null)
+        {
+            byte[] payloadBytes = new byte[data.sizeof()];
+            data.value().getBytes(0, payloadBytes);
```

Comment:
I think we have proposed code in other feedback comments for getting the byteArray vs byteBuffer and using those directly instead of allocating a byte[] to make a copy, right?